compiler/test/dotty/tools/utils.scala

```diff
@@ -43,3 +43,24 @@ def assertThrows[T <: Throwable: ClassTag](p: T => Boolean)(body: => Any): Unit
     case failed: T       => throw AssertionError(s"Exception failed check: $failed").tap(_.addSuppressed(failed))
     case NonFatal(other) => throw AssertionError(s"Wrong exception: expected ${implicitly[ClassTag[T]]} but was ${other.getClass.getName}").tap(_.addSuppressed(other))
 end assertThrows
+
+def toolArgsFor(files: List[JPath])(using codec: Codec = Codec.UTF8): List[String] =
+  files.flatMap(path => toolArgsParse(readLines(path.toFile)))
+
+// Inspect the first 10 of the given lines for compiler options of the form
+// `// scalac: args`, `/* scalac: args`, ` * scalac: args`.
+// If args string ends in close comment, drop the `*` `/`.
+// If split, parse the args string as a command line.
+// (from scala.tools.partest.nest.Runner#toolArgsFor)
+def toolArgsParse(lines: List[String]): List[String] = {
+  val tag  = "scalac:"
+  val endc = "*" + "/"    // be forgiving of /* scalac: ... */
+  def stripped(s: String) = s.substring(s.indexOf(tag) + tag.length).stripSuffix(endc)
+  val args = lines.to(LazyList).take(10).filter { s =>
+       s.contains("// " + tag)
+    || s.contains("/* " + tag)
+    || s.contains(" * " + tag)
+    // but avoid picking up comments like "% scalac ./a.scala" and "$ scalac a.scala"
```

Comment:
As a tester, I would like to use `//scalac: args`.

Reply:
OK, I'll add that.